src/main/java/com/github/mtsochacki/adventofcode/Day18.java

```diff
@@ -108,65 +106,63 @@ private List<RegularNumber> addSnailfish(List<RegularNumber> firstSnailfish, Lis
+                .filter(i -> snailfishNumber.get(i).getLevel() > 4)
+                .findFirst();
     }
 
-    private boolean isInNeedOfSplit(List<RegularNumber> snailfishNumber) {
-        return snailfishNumber.stream()
-                .anyMatch(regularNumber -> regularNumber.getValue() > 9);
+    private OptionalInt findIndexToSplit(List<RegularNumber> snailfishNumber) {
+        return IntStream.range(0, snailfishNumber.size())
+                .filter(i -> snailfishNumber.get(i).getValue() > 9)
+                .findFirst();
     }
 
-    private void explode(List<RegularNumber> snailfishNumber) {
-        for (int i = 0; i < snailfishNumber.size(); i++) {
-            if (snailfishNumber.get(i).getLevel() >= 5) {
-                if (i == 0) {
-                    snailfishNumber.get(i).setValue(0);
-                    snailfishNumber.get(i).decrementLevel();
-                    snailfishNumber.get(i + 2).increaseValue(snailfishNumber.get(i + 1).getValue());
-                    snailfishNumber.remove(i + 1);
-                } else if (i + 2 >= snailfishNumber.size()) {
-                    snailfishNumber.get(i - 1).increaseValue(snailfishNumber.get(i).getValue());
-                    snailfishNumber.get(i).setValue(0);
-                    snailfishNumber.get(i).decrementLevel();
-                    snailfishNumber.remove(i + 1);
-                } else {
-                    snailfishNumber.get(i - 1).increaseValue(snailfishNumber.get(i).getValue());
-                    snailfishNumber.get(i + 2).increaseValue(snailfishNumber.get(i + 1).getValue());
-                    snailfishNumber.get(i + 1).setValue(0);
-                    snailfishNumber.get(i + 1).decrementLevel();
-                    snailfishNumber.remove(i);
-                }
-                break;
-            }
+    private void explode(List<RegularNumber> snailfishNumber, int indexToExplode) {
+        if (indexToExplode == 0) {
+            snailfishNumber.get(indexToExplode).setValue(0);
+            snailfishNumber.get(indexToExplode).decrementLevel();
+            snailfishNumber.get(indexToExplode + 2).increaseValue(snailfishNumber.get(indexToExplode + 1).getValue());
+            snailfishNumber.remove(indexToExplode + 1);
+        } else if (indexToExplode + 2 >= snailfishNumber.size()) {
+            snailfishNumber.get(indexToExplode - 1).increaseValue(snailfishNumber.get(indexToExplode).getValue());
+            snailfishNumber.get(indexToExplode).setValue(0);
+            snailfishNumber.get(indexToExplode).decrementLevel();
+            snailfishNumber.remove(indexToExplode + 1);
+        } else {
+            snailfishNumber.get(indexToExplode - 1).increaseValue(snailfishNumber.get(indexToExplode).getValue());
+            snailfishNumber.get(indexToExplode + 2).increaseValue(snailfishNumber.get(indexToExplode + 1).getValue());
+            snailfishNumber.set(indexToExplode, new RegularNumber(4, 0));
```

Comment:
Is it guaranteed that this will always be level 4? It's not trivial to me, I would either include a comment explaining why that's the case or get the level from previous value